### PR TITLE
feat(patient): add panel to display useful patient information

### LINKED ISSUE #2259: Update patient profile page to always display useful information
## 🚀 Feature Proposal
The patient profile should always display the patient's most important information regardless of what tab they are on. 

The goal of this issue is to make a screen that looks similar to this: 
image

## Notes
- don't worry about colors or matching the mockup exactly, this is simply to get the layout in place
- for sections that that do not yet exist (i.e. procedures), don't worry about adding them
- In the allergies box, each item should link to the view allergies page
- for now, the diagnoses section should just display  a list of diagnoses and the date that diagnosis was on (don't worry about status/ICD-10 or secondary diagnoses since we don't have that concept yet).
- for now, ignore the new visit button, since the concept of visits does not yet exist

### PR DESCRIPTION
**Pull request by AlexTan331**
fix #2259 

**Changes proposed in this pull request:**

- add panel to display useful patient information

### CODE REVIEW

**Comment by michaelkramer** [comment]
@AlexTan331 I have downloaded and ran your branch.  Would you be open for some critiques?
I can send a more detailed list if you are interested.

1. Mobile view is bad
2. The `add allergies` button is miss placed.
image
3. on file `src/patients/view/ViewPatient.tsx` there is lots of repeat code.  Making an Array and then `map` through would allow for quicker edits in the future.

I have more, but only if you're interested.

**Comment by AlexTan331** [comment]
@michaelkramer  Thank you for your critiques. 
This is the my first attempt that successfully made the overall layout in place. It is obviously not ideal, and the reason why I didn't mark it as draft is because I am open to anyone just like you to provide useful suggestion/critiques so that I know if I am on the right path. 
So please share your opinions.

File: src/__tests__/patients/view/ImportantPatientInfo.test.tsx [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+import { createMemoryHistory } from 'history'
+import React from 'react'
+// import { startOfDay, subYears } from 'date-fns'
+import { act } from 'react-dom/test-utils'
+import { Provider } from 'react-redux'
+import { Router } from 'react-router-dom'
+import createMockStore from 'redux-mock-store'
+import thunk from 'redux-thunk'
+
+// import Diagnoses from '../../../patients/diagnoses/Diagnoses'
```
Comment (blestab) [inline src/__tests__/patients/view/ImportantPatientInfo.test.tsx:14]: May we remove the commented out code if it no longer being used

File: src/__tests__/patients/view/ImportantPatientInfo.test.tsx [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    })
+
+    it('should not render new diagnosis button if user does not have permissions', async () => {
+      const wrapper = await setup(expectedPatient, [])
+
+      expect(wrapper.find(components.Button)).toHaveLength(0)
+    })
+  })
+
+  //   describe('add new care plan button', () => {
```
Comment (blestab) [inline src/__tests__/patients/view/ImportantPatientInfo.test.tsx:265]: Do we still need this here?

File: src/patients/view/ImportantPatientInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+import Allergy from '../../shared/model/Allergy'
+import Diagnosis from '../../shared/model/Diagnosis'
+import Patient from '../../shared/model/Patient'
+import Permissions from '../../shared/model/Permissions'
+import { RootState } from '../../shared/store'
+import NewAllergyModal from '../allergies/NewAllergyModal'
+import AddCarePlanModal from '../care-plans/AddCarePlanModal'
+import AddDiagnosisModal from '../diagnoses/AddDiagnosisModal'
+import AddVisitModal from '../visits/AddVisitModal'
+// import  {getPatientFullName} from '../util/patient-name-util'
```
Comment (blestab) [inline src/patients/view/ImportantPatientInfo.tsx:17]: Remove commented out code

**Review by blestab — CHANGES_REQUESTED** [review]
Looks like we have a few lines of code commented out. Perhaps we should remove those if there is no plan to use them anymore.

**Comment by tehKapa** [comment]
@AlexTan331 there are conflicts to resolve

File: src/patients/view/ViewPatient.tsx [hunk 4]
```diff
@@ -28,6 +28,7 @@ import Note from '../notes/NoteTab'
 import RelatedPerson from '../related-persons/RelatedPersonTab'
 import { getPatientCode, getPatientFullName } from '../util/patient-util'
```
Comment (tehKapa) [inline src/patients/view/ViewPatient.tsx:29]: 'getPatientCode' is defined but never used

**Review by tehKapa — CHANGES_REQUESTED** [review]

**Review by tehKapa — APPROVED** [review]

**Comment by morrme** [comment]
LGTM

**Pull request merged**